batch/sql/estimated-current.sql

```diff
@@ -238,46 +243,50 @@ CREATE TABLE IF NOT EXISTS `batch_updates` (
 ) ENGINE = InnoDB;
 CREATE INDEX `batch_updates_committed` ON `batch_updates` (`batch_id`, `committed`);
 CREATE INDEX `batch_updates_start_job_id` ON `batch_updates` (`batch_id`, `start_job_id`);
+CREATE INDEX `batch_updates_start_job_group_id` ON `batch_updates` (`batch_id`, `start_job_group_id`);
 
 CREATE TABLE IF NOT EXISTS `job_groups_n_jobs_in_complete_states` (
   `id` BIGINT NOT NULL,
-  `job_group_id` INT NOT NULL DEFAULT 0,
+  `job_group_id` INT NOT NULL,
   `n_completed` INT NOT NULL DEFAULT 0,
   `n_succeeded` INT NOT NULL DEFAULT 0,
   `n_failed` INT NOT NULL DEFAULT 0,
   `n_cancelled` INT NOT NULL DEFAULT 0,
-  PRIMARY KEY (`id`),
-  FOREIGN KEY (`id`) REFERENCES batches(id) ON DELETE CASCADE
+  PRIMARY KEY (`id`, `job_group_id`),
+  FOREIGN KEY (`id`) REFERENCES batches(id) ON DELETE CASCADE,
+  FOREIGN KEY (`id`, `job_group_id`) REFERENCES job_groups (`batch_id`, `job_group_id`) ON DELETE CASCADE
 ) ENGINE = InnoDB;
 
 CREATE TABLE IF NOT EXISTS `job_groups_cancelled` (
   `id` BIGINT NOT NULL,
-  `job_group_id` INT NOT NULL DEFAULT 0,
-  PRIMARY KEY (`id`),
-  FOREIGN KEY (`id`) REFERENCES batches(id) ON DELETE CASCADE
+  `job_group_id` INT NOT NULL,
+  PRIMARY KEY (`id`, `job_group_id`),
+  FOREIGN KEY (`id`) REFERENCES batches(id) ON DELETE CASCADE,
+  FOREIGN KEY (`id`, `job_group_id`) REFERENCES job_groups (`batch_id`, `job_group_id`) ON DELETE CASCADE
 ) ENGINE = InnoDB;
 
 CREATE TABLE IF NOT EXISTS `job_groups_inst_coll_staging` (
   `batch_id` BIGINT NOT NULL,
   `update_id` INT NOT NULL,
-  `job_group_id` INT NOT NULL DEFAULT 0,
+  `job_group_id` INT NOT NULL,
   `inst_coll` VARCHAR(255),
   `token` INT NOT NULL,
   `n_jobs` INT NOT NULL DEFAULT 0,
   `n_ready_jobs` INT NOT NULL DEFAULT 0,
   `ready_cores_mcpu` BIGINT NOT NULL DEFAULT 0,
-  PRIMARY KEY (`batch_id`, `update_id`, `inst_coll`, `token`),
+  PRIMARY KEY (`batch_id`, `update_id`, `job_group_id`, `inst_coll`, `token`),
```

Comment:
The docs note that changing the primary key is "an expensive operation" https://dev.mysql.com/doc/refman/8.0/en/innodb-online-ddl-operations.html. No further details though. We should just be cognizant that some CPU might be used for the schema change which takes away from MJC, etc.